### PR TITLE
Add api acceptance tests for auto accept federated share

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Add acceptance tests for auto accept trusted servers and auto add server feature of federated sharing.

## Related Issue

- Related Issue https://github.com/owncloud/QA/issues/22

## How Has This Been Tested?

:robot: 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

**Review by haribhandari07 — COMMENTED** [review]
LGTM

File: tests/acceptance/features/apiFederation/federated.feature [hunk 1]
```diff
@@ -481,3 +481,75 @@ Feature: federated
...
+    And user "user1" from server "LOCAL" has accepted the last pending share
+    And using server "LOCAL"
+    Then url "%remote_server%" should be a trusted server
+    When user "user0" from server "REMOTE" shares "/textfile1.txt" with user "user1" from server "LOCAL" using the sharing API
+    # Uncomment this line once the issue is resolved
+    # Then as "user1" file "textfile1 (2).txt" should exist
+    Then as "user1" file "textfile1 (2).txt" should not exist
+
+  @issue-35839
+  Scenario: disable Add server automatically once a federated share was created successfully
```
Comment (individual-it) [inline tests/acceptance/features/apiFederation/federated.feature]: a bit confusing, maybe use quotes to make it clearer

**Review by individual-it — APPROVED** [review]

**Comment by dpakach** [comment]
Backport on #35843

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
Restarted drone - master CI should be fixed now.

**Pull request merged**